### ISSUE TITLE
Exception: No Redis password in environment variables!

### ISSUE THREAD

**Issue by cmungall** [biocypher/ChatGSE#12]
This looks like a great project!

When I follow the instructions for the streamlit app in the README I get an exception 

```
    def __init__(self):
        REDIS_PW = os.getenv("REDIS_PW")
        if not REDIS_PW:
            raise Exception("No Redis password in environment variables!")
        self.db = redis.Redis(
            host="redis-10494.c250.eu-central-1-1.ec2.cloud.redislabs.com",
            port=10494,
            password=REDIS_PW,
        )
        self.config = {}
```

This looks like a private redis instance, not clear if I should stand up my own.

setting a fake password allows me to advance with the app until it starts to process, then I get

```
Assistant: Thank you for the input. The model will be with you shortly. Please enter your questions below. These can be general, such as 'explain these results', or specific. General questions will yield more general answers, while specific questions go into more detail. You can follow up on the answers with more questions.

AuthenticationError: invalid username-password pair
Traceback:
File "/Users/cjm/Library/Caches/pypoetry/virtualenvs/chatgse-NErwA5td-py3.10/lib/python3.10/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 565, in _run_script
    exec(code, module.__dict__)
File "/Users/cjm/repos/ChatGSE/app.py", line 1078, in <module>
    main()
File "/Users/cjm/repos/ChatGSE/app.py", line 881, in main
    ss.response, ss.token_usage = cg._get_response()
File "/Users/cjm/repos/ChatGSE/chatgse/_interface.py", line 446, in _get_response
    response, token_usage, correction = ss.conversation.query(ss.input)
File "/Users/cjm/repos/ChatGSE/chatgse/_llm_connect.py", line 114, in query
    msg, token_usage = self._primary_query()
File "/Users/cjm/repos/ChatGSE/chatgse/_llm_connect.py", line 187, in _primary_query
    self._update_usage_stats(self.model, token_usage)
File "/Users/cjm/repos/ChatGSE/chatgse/_llm_connect.py", line 221, in _update_usage_stats
    self.usage_stats.increment(
File "/Users/cjm/Library/Caches/pypoetry/virtualenvs/chatgse-NErwA5td-py3.10/lib/python3.10/site-packages/decorator.py", line 232, in fun
    return caller(func, *(extras + args), **kw)
File "/Users/cjm/Library/Caches/pypoetry/virtualenvs/chatgse-NErwA5td-py3.10/lib/python3.10/site-packages/retry/api.py", line 73, in retry_decorator
    return __retry_internal(partial(f, *args, **kwargs), exceptions, tries, delay, max_delay, backoff, jitter,
File "/Users/cjm/Library/Caches/pypoetry/virtualenvs/chatgse-NErwA5td-py3.10/lib/python3.10/site-packages/retry/api.py", line 33, in __retry_internal
    return f()
File "/Users/cjm/repos/ChatGSE/chatgse/_stats.py", line 45, in increment
    self.db.zincrby(key, val, member)
File "/Users/cjm/Library/Caches/pypoetry/virtualenvs/chatgse-NErwA5td-py3.10/lib/python3.10/site-packages/redis/commands/core.py", line 4148, in zincrby
    return self.execute_command("ZINCRBY", name, amount, value)
File "/Users/cjm/Library/Caches/pypoetry/virtualenvs/chatgse-NErwA5td-py3.10/lib/python3.10/site-packages/redis/client.py", line 1255, in execute_command
    conn = self.connection or pool.get_connection(command_name, **options)
File "/Users/cjm/Library/Caches/pypoetry/virtualenvs/chatgse-NErwA5td-py3.10/lib/python3.10/site-packages/redis/connection.py", line 1442, in get_connection
    connection.connect()
File "/Users/cjm/Library/Caches/pypoetry/virtualenvs/chatgse-NErwA5td-py3.10/lib/python3.10/site-packages/redis/connection.py", line 710, in connect
    self.on_connect()
File "/Users/cjm/Library/Caches/pypoetry/virtualenvs/chatgse-NErwA5td-py3.10/lib/python3.10/site-packages/redis/connection.py", line 754, in on_connect
    auth_response = self.read_response()
File "/Users/cjm/Library/Caches/pypoetry/virtualenvs/chatgse-NErwA5td-py3.10/lib/python3.10/site-packages/redis/connection.py", line 865, in read_response
    response = self._parser.read_response(disable_decoding=disable_decoding)
File "/Users/cjm/Library/Caches/pypoetry/virtualenvs/chatgse-NErwA5td-py3.10/lib/python3.10/site-packages/redis/connection.py", line 346, in read_response
    result = self._read_response(disable_decoding=disable_decoding)
File "/Users/cjm/Library/Caches/pypoetry/virtualenvs/chatgse-NErwA5td-py3.10/lib/python3.10/site-packages/redis/connection.py", line 369, in _read_response
    raise error
```

**Comment by slobentanzer** [comment]
Hi @cmungall,
thanks for the report!

The redis DB is just for managing our community key volume; you should not have to use that locally when providing your own key. I missed some checks when implementing it, should be fixed now (`0.2.17`). :)

**Comment by slobentanzer** [comment]
Closing as it appears to be addressed. @cmungall feel free to reopen if you still encounter problems.
